File: shared/src/commonMain/kotlin/com/jetbrains/swiftinteropplayground/classes/SealedClass.kt
package com.jetbrains.swiftinteropplayground.classes

sealed class SealedClass {
    object Object : SealedClass()

    class Simple(val param1: String) : SealedClass()

    data class Data(val param1: String, val param2: Boolean) : SealedClass()
}

private fun example(s: SealedClass) {
    when (s) {
        SealedClass.Object -> TODO()
        is SealedClass.Simple -> TODO()
        is SealedClass.Data -> TODO()
    }
}

File: kotlin-swift-interopedia-samples-skie/iosApp/iosApp/Classes/SealedClass.swift
import Foundation
import shared

class SealedClassExample {
    func example(s: SealedClass) {
        switch onEnum(of: s) {
            case .object: print("object")
        case .simple(let simple): print("simple \(simple.param1)")
        case .data(let data): print("data \(data.param1) \(data.param2)")
        }
    }
}
